src/test/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfigurationTest.java

```diff
@@ -43,8 +60,48 @@
     public void setup() {
         pluginConfiguration = new BitbucketPluginConfiguration();
         when(validServerConfiguration.validate()).thenReturn(FormValidation.ok());
+        when(validServerConfiguration.getId()).thenReturn("0");
+        when(validServerConfiguration.getBaseUrl()).thenReturn("http://localhost:5990/bitbucket");
         when(invalidServerConfigurationOne.validate()).thenReturn(FormValidation.error(ERROR_MESSAGE));
+        when(invalidServerConfigurationOne.getId()).thenReturn("1");
+        when(invalidServerConfigurationOne.getBaseUrl()).thenReturn("http://localhost:6990/bitbucket");
         when(invalidServerConfigurationTwo.validate()).thenReturn(FormValidation.error(ERROR_MESSAGE));
+        when(invalidServerConfigurationTwo.getId()).thenReturn("2");
+        when(invalidServerConfigurationTwo.getBaseUrl()).thenReturn("http://localhost:7990/bitbucket");
+    }
+
+    @Test
+    public void testConfigureChangedBaseUrlUpdatesJob() throws Exception {
```

Comment:
I know the distinction between unit and integration tests is a bit fuzzy in our code, but this feels to me more like an integration test than a unit test and might be happier in `BitbucketPluginConfigurationTestIT`. 
It would be good have a few unit tests covering each job independently- which you can do if you use the `JenkinsProvider` in your method.
Also, not your code, but that means we could remove the `JenkinsRule` from this class, which would be nice.